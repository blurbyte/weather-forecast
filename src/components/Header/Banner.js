import styled from 'styled-components';

const Banner = styled.header`
  width: 100%;
  height: 6rem;
  border-bottom: 1px solid #222222;
`;

export default Banner;
