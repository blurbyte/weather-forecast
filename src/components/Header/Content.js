import styled from 'styled-components';

import BaseContent from '../Content';

const Content = styled(BaseContent)`
  display: flex;
  align-items: center;
`;

export default Content;
