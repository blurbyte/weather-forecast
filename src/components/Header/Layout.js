import styled from 'styled-components';

import { simpleFadeIn } from '../../styles/animations';

const Layout = styled.header`
  width: 100%;
  height: 6rem;
  border-bottom: 1px solid ${props => props.theme.primary};
  grid-area: header;
  animation: ${simpleFadeIn} 1.6s ease-out forwards;
  opacity: 0;
  animation-delay: 0.2s;
  z-index: 9;
`;

export default Layout;
