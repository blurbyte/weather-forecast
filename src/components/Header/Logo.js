import React from 'react';

const Logo = () => (
  <svg xmlns="http://www.w3.org/2000/svg" width="179" height="17">
    <path
      fill="currentColor"
      d="M6.889 15.872h2.376v.72H0v-.72h1.608V1.302H0v-.72h14.161v4.417h-.816c-.432-2.544-2.088-3.745-5.112-3.697H6.889v6.913h.816c1.848-.024 2.665-.912 2.665-2.88h.816v6.528h-.816c-.144-2.136-.888-2.952-2.736-2.928h-.745v6.937zM36.892 2.935c1.464 1.584 2.136 3.384 2.136 5.761 0 4.68-3.768 8.305-8.617 8.305-2.52 0-4.848-1.008-6.48-2.76-1.44-1.56-2.136-3.409-2.136-5.665 0-2.232.72-4.152 2.136-5.641 1.68-1.776 3.984-2.76 6.457-2.76 2.544-.001 4.848.984 6.504 2.76zM53.939 15.872h1.68v.72H47.05v-.72h1.608V1.302H47.05v-.72h8.977c2.712 0 3.504.096 4.584.528 1.272.504 2.064 1.728 2.064 3.24 0 1.44-.744 2.616-1.992 3.192-.768.36-1.32.48-2.88.672 3.048.432 4.344 1.705 4.369 4.272v.96c.024 1.392.024 1.488.144 1.728.144.336.408.504.744.504.264 0 .48-.048.96-.264l.312.456c-1.272.818-2.256 1.13-3.48 1.13s-2.352-.408-3-1.104c-.672-.672-.96-1.704-.96-3.264v-1.368c0-2.064-.528-2.784-1.968-2.784h-.984v7.392zm1.225-8.113c1.752 0 2.28-.744 2.28-3.216 0-2.497-.528-3.241-2.28-3.241H53.94v6.457h1.224zM74.066 1.302h-1.608v-.72h14.354v4.152h-.816c-.552-2.4-2.208-3.504-5.136-3.433h-1.512v6.913h.816c1.848-.024 2.665-.912 2.665-2.88h.816v6.528h-.816c-.144-2.136-.888-2.952-2.736-2.928h-.744v6.937h2.136c3 .096 4.536-1.296 5.088-4.608h.792v5.329H72.458v-.72h1.608V1.302zM110.737 16.592h-.792l-.504-1.296c-1.873 1.368-2.88 1.704-4.945 1.704-5.136 0-8.785-3.409-8.785-8.233 0-5.017 3.648-8.593 8.785-8.593 1.968 0 3.145.408 4.489 1.56l.672-1.152h.792l.072 4.656h-.792c-.288-.96-.48-1.416-.816-2.016-.792-1.392-2.232-2.256-3.768-2.256-2.424 0-3.745 2.016-3.745 5.737v3.84c0 1.896.216 2.976.792 3.984.624 1.104 1.68 1.68 3.048 1.68 1.536 0 2.712-.648 3.792-2.064.672-.888.984-1.512 1.296-2.64h.792l-.383 5.089zM122.734 12.271l-.984 2.616c-.048.168-.096.336-.096.456 0 .384.288.528 1.008.528h.648v.72h-4.656v-.72h.576c1.008-.024 1.416-.36 1.848-1.512l5.4-13.777h1.824l6.264 15.289h1.368v.72h-8.688v-.72h1.68l-1.44-3.6h-4.752zm4.464-.72l-2.136-5.232-2.041 5.232h4.177zM155.169 4.951h-.792c-.288-.912-.528-1.416-.96-1.968-.96-1.272-2.496-2.016-4.152-2.016-1.56 0-2.52.696-2.52 1.8 0 1.008.696 1.656 2.376 2.184l2.904.936c2.976.936 4.128 2.304 4.128 4.873 0 3.672-2.521 6.241-6.145 6.241-1.872 0-3.24-.552-4.656-1.848l-.768 1.44h-.792v-5.088h.792c.312.984.504 1.464.864 2.064.984 1.632 2.736 2.641 4.608 2.641 1.896 0 3.168-1.008 3.168-2.473 0-1.176-.696-1.896-2.352-2.448l-3-1.008c-2.88-.984-4.2-2.448-4.2-4.608 0-1.32.504-2.664 1.368-3.672 1.056-1.224 2.472-1.824 4.272-1.824 1.848 0 3.168.504 4.345 1.704l.72-1.296h.792v4.366zM173.762 15.872h2.28v.72h-9.841v-.72h2.28V1.302h-.72c-1.68.048-2.712 1.464-3.24 4.393h-.792l.288-5.112h14.209l.288 5.112h-.816c-.504-2.928-1.536-4.344-3.216-4.393h-.72v14.57z"
    />
  </svg>
);

export default Logo;
