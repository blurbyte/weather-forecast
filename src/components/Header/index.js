import React from 'react';
import PropTypes from 'prop-types';

import Search from '../Search';
import Content from './Content';
import Layout from './Layout';
import Logo from './Logo';

const Header = ({ fetchForecast }) => (
  <Layout>
    <Content>
      <Logo />
      <Search fetchForecast={fetchForecast} />
    </Content>
  </Layout>
);

Header.propTypes = {
  fetchForecast: PropTypes.func
};

export default Header;
