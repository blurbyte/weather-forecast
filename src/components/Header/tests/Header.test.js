import React from 'react';
import renderer from 'react-test-renderer';

import Header from '../index';

jest.mock('../Logo', () => 'Logo');
jest.mock('../../SearchCity', () => 'SearchCity');

test('renders correctly', () => {
  const component = renderer.create(<Header fetchForecast={jest.fn()} />);
  expect(component.toJSON()).toMatchSnapshot();
});
