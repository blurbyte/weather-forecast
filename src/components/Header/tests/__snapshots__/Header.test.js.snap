// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<header
  className="sc-htpNat frKZNy"
>
  <div
    className="sc-bwzfXH jaPkJ sc-bdVaJa hNXGXU"
  >
    <Logo />
    <Search
      fetchForecast={undefined}
    />
  </div>
</header>
`;
